tests/spec/README.txt

```diff
@@ -0,0 +1,6 @@
+These tests specify the expected behaviour of ehrQL, and by extension, the QM.
```

Comment:
Following Iain, I'll put general thoughts on this approach to testing here. Multiple comments to facilitate threaded conversation.